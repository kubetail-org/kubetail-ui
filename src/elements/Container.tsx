// Copyright 2024 Andres Morey
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { cn } from '@/lib/utils';
import type { ElementType, ReactNode } from 'react';

const baseCN = 'container mx-auto sm:max-w-screen-sm md:max-w-screen-md lg:max-w-screen-lg xl:max-w-screen-xl 2xl:max-w-screen-2xl px-4';

type Props = {
  as?: ElementType;
  children?: ReactNode;
  className?: string;
};

const Container = ({ as = 'div', children, className }: Props) => {
  const Tag = as;

  return (
    <Tag className={cn(baseCN, className)}>
      {children}
    </Tag>
  );
};

export default Container;
