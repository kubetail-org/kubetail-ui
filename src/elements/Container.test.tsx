// Copyright 2024 Andres Morey
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

import { render } from '@testing-library/react';

import Container from './Container';

describe('Container', () => {
  it('renders contents properly', () => {
    const { asFragment } = render(<Container>contents</Container>);
    expect(asFragment()).toMatchSnapshot();
  });

  it('appends custom className', () => {
    const { asFragment } = render(<Container className="text-red-100" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it('renders custom tagName properly', () => {
    const { asFragment } = render(<Container as="header" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
