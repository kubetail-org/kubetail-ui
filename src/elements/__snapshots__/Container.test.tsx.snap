// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Container > appends custom className 1`] = `
<DocumentFragment>
  <div
    class="container mx-auto sm:max-w-screen-sm md:max-w-screen-md lg:max-w-screen-lg xl:max-w-screen-xl 2xl:max-w-screen-2xl px-4 text-red-100"
  />
</DocumentFragment>
`;

exports[`Container > renders contents properly 1`] = `
<DocumentFragment>
  <div
    class="container mx-auto sm:max-w-screen-sm md:max-w-screen-md lg:max-w-screen-lg xl:max-w-screen-xl 2xl:max-w-screen-2xl px-4"
  >
    contents
  </div>
</DocumentFragment>
`;

exports[`Container > renders custom tagName properly 1`] = `
<DocumentFragment>
  <header
    class="container mx-auto sm:max-w-screen-sm md:max-w-screen-md lg:max-w-screen-lg xl:max-w-screen-xl 2xl:max-w-screen-2xl px-4"
  />
</DocumentFragment>
`;
